src/lib.rs

```diff
@@ -46,17 +47,29 @@ widget_ids!(pub struct Ids {
     longitude_text[],
     filer_button[],
     airports[],
-    planes[]
+    planes[],
+    square
 });
 
 pub use util::MAP_PERF_DATA;
 
+#[derive(Copy, Clone)]
+pub struct Vertex {
+    pub position: [f32; 2],
+    pub angle: f32,
+    pub offset: [f32; 2],
+    pub dpi_factor: f32,
```

Comment:
Would this work better as a uniform?

Reply:
I moved dpi_factor to be a part of the uniform. Angle and offset are specific to each plane, so I am not sure how those would be part of the uniform. The position couldn't really be either since it is per-vertex, unless I am misunderstanding how uniforms work.